### ISSUE TITLE
Projections.id() should be used instead of DocumentBuilderIndexedEntity.getIdPropertyName()

### ISSUE THREAD

**Issue by yrodiere** [mincong-h/gsoc-hsearch#165]
The current implementation assumes that the ID field name can always be retrieved through `org.hibernate.search.engine.spi.DocumentBuilderIndexedEntity.getIdPropertyName()`.

This is not true, as this method returns null when provided IDs are used (this is the case in the Infinispan integration of Hibernate Search for instance). `Projections.id()` should be used instead.

See:

 * https://github.com/mincong-h/gsoc-hsearch/blob/21be31a72a59e933006c1ccdc9e372b76a979098/core/src/main/java/org/hibernate/search/jsr352/internal/steps/lucene/EntityReader.java#L245
 * https://github.com/mincong-h/gsoc-hsearch/blob/81a39117cb8f9c8d0086c2252c2a3c759488fa4f/core/src/main/java/org/hibernate/search/jsr352/internal/util/MassIndexerUtil.java#L29

**Comment by mincong-h** [comment]
Thanks for pointing out this. I just saw the constructor of class `DocumentBuilderIndexedEntity` which assigns the id name using the following line:

```java
this.identifierName = idProvided ? null : idPropertyMetadata.getPropertyAccessor().getName();
```

**Comment by yrodiere** [comment]
Solved in #167
